enhancements/authentication/unsupported-direct-use-of-oidc.md

```diff
@@ -0,0 +1,368 @@
+Read more in the [attached study](./direct-oidc-study/study-oidc-in-openshift.md).
+
+In order to allow proper integration with a 3rd party OIDC provider, additional
+configuration API would have to exist. It would have to include:
+1. a reference to the OAuth2 client credentials that the console can use
+2. configuration of the scopes it should use to request authorization from the
+   3rd party IdP
+
+Additional considerations:
+1. The console would likely have to forward the ID tokens it retrieves to the
+   kube-apiserver to get the `userInfo` based on the kube-apiserver configuration,
+   not on the configuration of the console to avoid discrepancies in console/kube-apiserver
+   identity configuration.
+2. From 1. we can see that the ID tokens for the console would likely have to contain
+   both the `kube-apiserver` and `console` OAuth2 clients in the audience claim
+3. ID tokens are not revocable which is commonly solved by these being short-lived.
+   The `console` should therefore implement token refresh.
+
+**Open question**: based on 3. we need refresh tokens for seamless web console use.
+During token refresh, an ID token does not necessarily have to be returned. There
+are a few options in that case:
+1. Ignore the above 1. and have the console authenticate the user itself and impersonate
+   it in the requests to the kube-apiserver.
+2. Use the access token from the token refresh flow to retrieve claims from the
+   `UserInfo` endpoint and forward those to the kube-apiserver. However, these
+   are not time-limited (no `exp` claim) and do not have to contain the `iss` and `aud` claims.
+3. Have the configuring user decide. Maybe ID tokens are present in token refresh
+   flow. Maybe at least `iss` and `aud` are present in the `UserInfo` endpoint responses.
+   The user should know.
+
+#### oc login
+
+`oc login`, like the web console, has many assumptions on its own:
+1. existence of the `openshift-challenging-client`
+2. the trust should be present either in the system bundle or in the CA bundle from the kubeconfig
+3. `oc` uses the issuer with added `/oauth/token/implicit` path as the `redirect_uri` for the implicit flow
+4. `oc` expects to be able to use HTTP challenge to authenticate to the authorization endpoint of the issuer
+
+In order to get `oc login` to work, [oc Authorization Grant Login](./improved-login-workflow.md)
+should get implemented. Additionally, it should be able to retrieve parameters
+specific to the OIDC provider deployment, such as which OAuth2 client to use
+and which scopes to request.
+
+As an alternative, `oc login` might try to use an exec plugin supplied by the
+OIDC provider directly instead of attempting something of its own. Perhaps `oc login`
+would not be used for cluster login at all.
+
+Even now, `oc` is capable of using an ID token directly if the kube-apiserver
+is configured accordingly. The main issue is with retrieving the token.
```

Comment:
> The main issue is with retrieving the token.

For which the kube community has solutions, i.e. https://github.com/int128/kubelogin. Maybe worth mentioning that this can also be used to craft a kubeconfig which `oc` can also use?

Reply:
added in the latest patch